### PR TITLE
e50hx: Initial implementation

### PR DESCRIPTION
**Pull request by jontrulson**
This driver requires the UPM BACNETMSTP driver (PR #385) to be merged
first.

This module implements support for the Veris E50H2 and E50H5
BACnet Energy Meters.

From the datasheet: The E50H5 BACnet MS/TP DIN Rail Meter with
Data Logging combines exceptional performance and easy
installation to deliver a cost-effective solution for power
monitoring applications. Native serial communication via BACnet
MS/TP provides complete accessibility of all measurements to your
Building Automation System. The data logging capability protects
data in the event of a power failure. The E50H5 can be easily
installed on standard DIN rail, surface mounted or contained in
an optional NEMA 4 enclosure, as needed. The front-panel LCD
display makes device installation and setup easy and provides
local access to the full set of detailed measurements.

This module was developed using the upm::BACNETMSTP module, based
on libbacnet-stack 0.8.3.  Both libbacnet 0.8.3 and the
upm::BACNETMSTP libraries must be present in order to build this
module.  This driver was developed on the E50H5.  The Trend Log
functionality is not currently supported.

The Binary Input Objects are also not supported as these are only
used for the Alarm bits which are already available from Analog
Input Object 52 as an alarm bitfield incorporating all of the
supported alarm indicators.

It was connected using an RS232->RS485 interface.  You cannot use
the built in MCU TTL UART pins for accessing this device -- you
must use a full Serial RS232->RS485 or USB-RS485 interface
connected via USB.

Signed-off-by: Jon Trulson <jtrulson@ics.com>

### CODE REVIEW

File: src/e50hx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+set (module_h ${libname}.h)
+
+pkg_search_module(BACNET libbacnet)
+if (BACNET_FOUND)
+  # upm-libbacnetmstp will bring in libbacnet, I hope
+  set (reqlibname "upm-bacnetmstp")
+  include_directories(${BACNET_INCLUDE_DIRS})
+  include_directories("../bacnetmstp")
+  upm_module_init()
+  add_dependencies(${libname} bacnetmstp)
```
Comment (Propanu) [inline src/e50hx/CMakeLists.txt]: This is not needed anymore, see #382.

Comment (jontrulson) [inline src/e50hx/CMakeLists.txt]: Yes, I've been reading through the MRAA and UPM logs, PR's and issues this morning to catch up.  Do you want me to fix this?

Comment (Propanu) [inline src/e50hx/CMakeLists.txt]: That would help yes!

**Comment by Propanu** [comment]
Merged, thanks!

**Pull request closed without merging**